data_structures/stack/stack.py

```diff
@@ -0,0 +1,48 @@
+from node import Node
+
+
+class Stack:
+    """Define stack object."""
+    def __init__(self, iter=[]):
+        self.top = None
+        self._size = 0
+        # self.items = []
+        try:
+            for item in iter:
+                self.push(item)
+        except TypeError:
+            print('Value error! Please insert an iterable.')
+
+    def __len__(self):
+        """Return the size of the stack."""
+        return self._size
+
+    def __repr__(self):
+        """Return the value of the top."""
+        try:
+            return '<top> => {}'.format(self.top.val)
+        except ValueError:
+            print('Oops! The top has no value.')
```

Comment:
Same issues with this implementation of `__repr__` as the implementation of `__str__` above.